ISPC devel leads to failed build

## Issue description

Since this commit: https://github.com/NixOS/nixpkgs/commit/3dea1972737f5ce7b2c5461fe20370bad10aae03 my tool based on ISPC is not building anymore, see this issue I posted upstream: 
https://github.com/ispc/ispc/issues/1292

The commit explains the reason of the use of llvm 4.0 instead of 3.9, but not the reason for the use of an unreleased ISPC version. It works for me to keep the llvm 4.0 but revert to the released ispc 1.9.1 by "overrideAttrs" `version`.

## Technical details

* System: NixOS, 18.03pre115705.c8e7aab0c8 (Impala)
* Nix version: nix-env (Nix) 1.11.15
* Nixpkgs version: 18.03pre-git I'm using nixpkgs master (45400d65b4d7b0f1460f03b5ac3e57e458d8c538 the 1 of october 2017)
* Sandboxing enabled: build-use-sandbox = false

Poking at this (see the above commit and branch), here's what I'm seeing when updating the LLVM version of ISPC without updating ispc itself:

```unpacking sources
unpacking source archive /nix/store/lhagy113ffcryhk3n6ikdha9lyic9ax3-ispc-v1.9.1-src
source root is ispc-v1.9.1-src
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/flb9ar1xdd13c606aa4my9miy3iv4vyk-bash-4.4-p12/bin/bash CXX=/nix/store/6ph6yzj72lz1gmj9bdq5ngqkqqqf9s4c-clang-wrapper-4.0.1/bin/clang++ CLANG=/nix/store/6ph6yzj72lz1gmj9bdq5ngqkqqqf9s4c-clang-wrapper-4.0.1/bin/clang CLANG_INCLUDE=/nix/store/24fm4ib7rgphz5w96hrmn251vlb4ni6q-clang-4.0.1/include
ldconfig: Can't open cache file /nix/store/zpg78y1mf0di6127q6r51kgx2q8cxsvv-glibc-2.25-49/etc/ld.so.cache
: No such file or directory
 ============================== !!! WARNING !!! =============================== 
 Location of LLVM files in your PATH is different than path in LLVM_HOME 
 variable (or LLVM_HOME is not set). The most likely this means that you are 
 using default LLVM installation on your system, which is very bad sign. 
 Note, that ISPC uses LLVM optimizer and is highly dependent on it. We recommend 
 using *patched* version of LLVM 3.8. Patches are availible in 
 llvm_patches folder. You can build LLVM manually, or run our scripts, which 
 will do all the work for you. Do the following: 
 1. Create a folder, where LLVM will reside and set LLVM_HOME variable to its 
   path. 
 2. Set ISPC_HOME variable to your ISPC location (probably current folder). 3. Run alloy.py tool to checkout and build LLVM: 
   alloy.py -b --version=3.8 
 4. Add $LLVM_HOME/bin-3.8/bin path to your PATH. 
 ==============================================================================
Updating dependencies
In file included from ast.cpp:40:
In file included from ./ast.h:41:
In file included from ./ispc.h:41:
./ispc_version.h:67:2: error: "Unhandled LLVM version"
#error "Unhandled LLVM version"
 ^
In file included from ast.cpp:40:
In file included from ./ast.h:41:
./ispc.h:44:2: error: "Only LLVM 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8 and 3.9 development branch are supported"
#error "Only LLVM 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8 and 3.9 development branch are supported"
 ^
2 errors generated.
Using LLVM 4.0.1 from /nix/store/i6srh22283wdhiphyahqw3kskrwp89rc-llvm-4.0.1-lib/lib
Using compiler to build: clang version 4.0.1 (tags/RELEASE_401/final)
Creating objs/ directory
Compiling ast.cpp
In file included from ast.cpp:40:
In file included from ./ast.h:41:
In file included from ./ispc.h:41:
./ispc_version.h:67:2: error: "Unhandled LLVM version"
#error "Unhandled LLVM version"
 ^
In file included from ast.cpp:40:
In file included from ./ast.h:41:
./ispc.h:44:2: error: "Only LLVM 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8 and 3.9 development branch are supported"
#error "Only LLVM 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8 and 3.9 development branch are supported"
 ^
2 errors generated.
make: *** [Makefile:287: objs/ast.o] Error 1
```

Which I think is the answer for why the ISPC version was updated as well.

This doesn't resolve your issue but hopefully sheds some light on the situation!

--- a/pkgs/development/compilers/ispc/default.nix
+++ b/pkgs/development/compilers/ispc/default.nix
@@ -3,8 +3,8 @@ testedTargets ? ["sse2" "host"] # the default test target is sse4, but that is n
 }:
 
 stdenv.mkDerivation rec {
-  version = "20170807";
-  rev = "6e0fc2f148e95afad998a7c7f4d7908d29fd8e44";
+  version = "1.9.2";
+  rev = "v${version}";
 
   inherit testedTargets;
 
@@ -14,7 +14,7 @@ stdenv.mkDerivation rec {
     owner = "ispc";
     repo = "ispc";
     inherit rev;
-    sha256 = "17fwnfm8a329lgfhjwcvji4h1fm4iqmc28wz23hvgqbpj8lk6qgh";
+    sha256 = "0zaw7mwvly1csbdcbz9j8ry89n0r1fag1m1f579l4mgg1x6ksqry";
   };
 
   # there are missing dependencies in the Makefile, causing sporadic build failures

